src/query/compiler.js

```diff
@@ -47,9 +47,7 @@ assign(QueryCompiler.prototype, {
       defaults.as = this.single.as;
     }
 
-    if(this.client && this.client.prepBindings) {
-      defaults.bindings = this.client.prepBindings(defaults.bindings);
-    }
+    defaults.bindings = this.client.prepBindings(defaults.bindings || []);
```

Comment:
Is here possible that this.client doesn't exist? e.g. in case `require('knex').raw('? ?', [undefined, 'wat']).toSQL();`